### PR TITLE
Changes to add ir-rgb frame

### PR DESCRIPTION
**Pull request by abhijitmajumdar**
Changes to make the Left IR (infra 1) frame to stream RGB instead of the normal grayscale.  This is verified to be working with `librealsense` version `2.19.1`

### CODE REVIEW

**Comment by 130s** [comment]
What makes this WIP?

**Comment by abhijitmajumdar** [comment]
> What makes this WIP?

- The CI fails, since I changes the infrared image to be a RGB 3 channel instead of a single channel, which is causing most of the problems
- Code is workable but is still a hacky code, refactoring it right now to be close to generic (or should I close it and add another PR for that?)

**Comment by 130s** [comment]
> - The CI fails, since I changes the infrared image to be a RGB 3 channel instead of a single channel, which is causing most of the problems

I looked at the [failing CI](https://travis-ci.org/plusone-robotics/realsense/builds/514801070#L5105). Though at a glance I can't tell if these are negligible, indeed the change in this PR seems to change some performance (so that one of the tests failed due to the resulted RGB figure didn't stay within the test criteria). 

Just for brevity I'm citing below the 5 test cases that failed:

```
points_cloud_1                FAILED : Expect average color of [ 125.  116.  102.] (+-2.000). Got average of [ 127.82555213  116.97886198  103.44018592].
align_depth_ir1_1             FAILED : 'num_channels'
static_tf_1                   FAILED : Tf is None for couple camera_infra1_frame->camera_color_frame
depth_decimation_1            FAILED : 'num_channels'
align_depth_ir1_decimation_1  FAILED : 'num_channels'
```

How we want to move on, I'd say:
- For POR's fork, I'm fine to merge as long as we'll continue iterate quickly to fix any potential issues that this change may cause.
- For sending to upstream in the very near future, we'll have to let CI pass. We *can* even update the test cases if we absolutely think the current test cases are not appropriate (at the moment I can't tell how good the current test cases are, so I have no idea on this).

**Comment by abhijitmajumdar** [comment]
> How we want to move on, I'd say:

I agree to both the points

File: realsense2_camera/src/base_realsense_node.cpp [hunk 1]
```diff
@@ -93,10 +93,10 @@ BaseRealSenseNode::BaseRealSenseNode(ros::NodeHandle& nodeHandle,
     _stream_name[RS2_STREAM_DEPTH] = "depth";
     _depth_aligned_encoding[RS2_STREAM_DEPTH] = sensor_msgs::image_encodings::TYPE_16UC1;
 
-    // Infrared stream
-    _image_format[RS2_STREAM_INFRARED] = CV_8UC1;    // CVBridge type
-    _encoding[RS2_STREAM_INFRARED] = sensor_msgs::image_encodings::MONO8; // ROS message type
-    _unit_step_size[RS2_STREAM_INFRARED] = sizeof(uint8_t); // sensor_msgs::ImagePtr row step size
+    // Infrared-rgb stream
```
Comment (shaun-edwards) [inline realsense2_camera/src/base_realsense_node.cpp:96]: Changing the infrared stream to to be 3 channel (color) is causing several [tests](https://travis-ci.org/plusone-robotics/realsense/builds/514801070#L5101) to fail.  An alternative approach would be to create a second stream OR add a "switch" to enable colorized IR.  By default it would run IR only, allowing the failing tests to pass.

File: realsense2_camera/src/base_realsense_node.cpp [hunk 2]
```diff
@@ -663,7 +663,7 @@ void BaseRealSenseNode::publishAlignedDepthToOthers(rs2::frameset frames, const
         if (RS2_STREAM_DEPTH == stream_type)
             continue;
 
-        auto stream_index = frame.get_profile().stream_index();
+        auto stream_index = (stream_type==RS2_STREAM_INFRARED)?1:frame.get_profile().stream_index();
```
Comment (shaun-edwards) [inline realsense2_camera/src/base_realsense_node.cpp:666]: Why is this required?  What if the stream type isn't INFRARED?

**Comment by 130s** [comment]
Merging as the physical testing passed with this in one of our applications.

Immediate things to do after the merge:
- I'll tag POR's temporary version out of `plusone-robotics:por_upstream_sync` branch (it's early to break POR's default por_development branch at this moment).
- I'll create another ticket to fix the failing CI.

File: realsense2_camera/src/base_realsense_node.cpp [hunk 3]
```diff
@@ -1315,7 +1337,8 @@ void BaseRealSenseNode::frame_callback(rs2::frame frame)
             {
                 auto f = (*it);
                 auto stream_type = f.get_profile().stream_type();
-                auto stream_index = f.get_profile().stream_index();
+                // auto stream_index = f.get_profile().stream_index();
+                auto stream_index = (stream_type==RS2_STREAM_INFRARED)?1:f.get_profile().stream_index();
```
Comment (shaun-edwards) [inline realsense2_camera/src/base_realsense_node.cpp:1341]: Why is this required? What if the stream type isn't INFRARED?

File: realsense2_camera/src/base_realsense_node.cpp [hunk 4]
```diff
@@ -1938,8 +1961,8 @@ void BaseRealSenseNode::publishFrame(rs2::frame f, const ros::Time& t,
     }
 
     ++(seq[stream]);
-    auto& info_publisher = info_publishers.at(stream);
     auto& image_publisher = image_publishers.at(stream);
+    auto& info_publisher = info_publishers.at(stream);
```
Comment (shaun-edwards) [inline realsense2_camera/src/base_realsense_node.cpp:1965]: Is this a real change or did the line just move?

File: realsense2_camera/src/base_realsense_node.cpp (hunk 1 above)
Comment (abhijitmajumdar) [reply realsense2_camera/src/base_realsense_node.cpp:96]: that is the intent for further commits
I could add another stream, but I feel that would have potential problems down the line
I like the idea to add a switch to change the IR to the RGB

File: realsense2_camera/src/base_realsense_node.cpp (hunk 2 above)
Comment (abhijitmajumdar) [reply realsense2_camera/src/base_realsense_node.cpp:666]: Since infrared needs the index to match [this](https://github.com/plusone-robotics/realsense/blob/por_upstream_sync/realsense2_camera/include/realsense_node_factory.h#L29) which specifies it as a `1`. However when this is indexed normally using the `frame.get_profile().stream_index()` it returns `0`.

I realize this is a hack for now, and will be fixed soon

File: realsense2_camera/src/base_realsense_node.cpp (hunk 3 above)
Comment (abhijitmajumdar) [reply realsense2_camera/src/base_realsense_node.cpp:1341]: [answered here](https://github.com/plusone-robotics/realsense/pull/21#discussion_r271960653)
